wire-schema/src/main/java/com/squareup/wire/schema/MarkSet.java

```diff
@@ -83,9 +83,17 @@ boolean mark(ProtoType type) {
   boolean mark(ProtoMember protoMember) {
     if (protoMember == null) throw new NullPointerException("type == null");
     if (identifierSet.excludes(protoMember)) return false;
-    return members.containsKey(protoMember.type())
+    boolean result = members.containsKey(protoMember.type())
         ? members.put(protoMember.type(), protoMember)
         : types.add(protoMember.type());
+
+    // An option can hide another one.
+    String enclosingTypeOrPackage = protoMember.type().enclosingTypeOrPackage();
+    if (enclosingTypeOrPackage != null && enclosingTypeOrPackage.equals("google.protobuf")) {
```

Comment:
```suggestion
    if ("google.protobuf".equals(enclosingTypeOrPackage)) {
```
Not sure how you feel about "Yoda" conditionals...

Pretty sure @swankjesse doesn't like them...

Reply:
We should just convert to Kotlin then who cares